libdrizzle/state.cc

```diff
@@ -86,6 +86,12 @@ drizzle_return_t drizzle_state_packet_read(drizzle_st *con)
 
   con->packet_size= drizzle_get_byte3(con->buffer_ptr);
 
+  if (con->buffer_size < con->packet_size)
```

Comment:
While testing I found that the error also occurred with `(con->buffer_size - con->packet_size) < 4`. Thus I suggest changing the if statement to this: 

```c 
if (con->buffer_size < (con->packet_size + 4))
```